src/main/java/com/kickstarter/dropwizard/metrics/influxdb/InfluxDbMeasurement.java

```diff
@@ -147,34 +152,85 @@ public Builder putTag(final String key, final String value) {
     /**
      * Adds the given key-value pair to the fields map.
      *
-     * @throws IllegalArgumentException if any value is not a String or primitive.
+     * @throws IllegalArgumentException if any value is not a String, primitive,
+     *         or Collection of Strings and primitives.
      */
     public <T> Builder putField(final String key, final T value) {
-      if (value == null) {
-        return this;
+      if (value instanceof Collection<?>) {
+        final Collection collection = (Collection) value;
+        final String collString = validatedPrimitiveCollection(key, collection);
+        fields.put(key, collString);
+      } else if (value != null) {
+        final Optional<String> fieldString = validatedPrimitiveField(key, value);
+        fieldString.ifPresent(s -> fields.put(key, s));
+      }
+
+      return this;
+    }
+
+    /**
+     * Validates that the collection only contains Strings and primitives.
+     *
+     * @return the collection as a string, if valid.
+     * @throws IllegalArgumentException if any collection value is not a string or primitive.
+     */
+    private static String validatedPrimitiveCollection(final String key, final Collection collection) {
+      for (final Object value : collection) {
+        if (!isValidField(value)) {
+          throw new IllegalArgumentException(
+            String.format(
+              "InfluxDbMeasurement collection field '%s' must contain only Strings and primitives: invalid field '%s'",
+              key, value
+            )
+          );
+        }
       }
 
+      return collection.toString();
+    }
+
+    /**
+     * Validates that the field is a String or primitive.
+     *
+     * @return true if the field is a String or primitive.
+     */
+    private static <T> boolean isValidField(final T value) {
+      return VALID_FIELD_CLASSES.stream().anyMatch(klass -> klass.isInstance(value));
```

Comment:
do we not want to do the checks here that we do below for `isNaN ` and `isInfinite`?  Not sure if those can be included in a collection...

Reply:
We filter out NaN and Inf when writing a single field since InfluxDB [doesn't support these numerical values](https://github.com/influxdata/influxdb/issues/4089), and will blow up if we insert them into the DB.

Since we're writing collections as strings, we can serialize whatever we want, including NaN/Inf/null; I think this is desired over invalidating the entire collection.

Even the class validations are unnecessary, but I want to avoid allowing people to serialize things like `[<SomeObject@0x12345>, <SomeObject@0x90210>]` (unless they explicitly call .toString()).

Reply:
cool 👌